src/utils/Helpers/ML/ML.js

```diff
@@ -50,19 +50,50 @@ const getParsedTransactions = (transactions, addresses) => {
         isConfirmed: transaction.isConfirmed,
       }
 
-    const isOutputMine = addresses.some(
-      (address) => transaction.outputs[0].destination === address,
+    const isInputMine = addresses.some(
+      (address) => transaction.inputs[0].utxo.destination === address,
     )
 
-    const direction = !isOutputMine ? 'out' : 'in'
-    const destAddress =
-      direction === 'in' && transaction.outputs.length > 1
-        ? transaction.outputs[1].destination
```

Comment:
not every transactions having from-to order, so I updated to compare inputs-outputs destinations with wallet addresses